### PR TITLE
Added -Wconversion to .bazelrc

### LINKED ISSUE #1514: Add -Wconversion flag to .bazelrc
### Description of the task
Will help catch more bugs in our code

### Acceptance criteria

- [ ] as titled
- [ ] Fix build errors

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
Added -Wconversion warning to .bazelrc to catch more errors and fixed resulting build failures

### Testing Done
CI

### Resolved Issues
resolves #1514 

### Length Justification
this was not fun...

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/firmware/app/control/trajectory_planner_impl_test.cpp [hunk 1]
```diff
@@ -51,15 +51,15 @@ TEST(TrajectoryPlannerImplTest, test_robot_state_generation_at_constant_t_interv
         segment_lengths_angular);
 
     const float delta_t = (path_parameters.t_end - path_parameters.t_start) /
-                          (path_parameters.num_elements - 1);
+                          (float)(path_parameters.num_elements - 1);
```
Comment (MathewMacDougall) [inline src/firmware/app/control/trajectory_planner_impl_test.cpp]: nit: we prefer `static_cast` in all `.cpp` files

File: src/firmware/app/control/trajectory_planner_test.cpp [hunk 2]
```diff
@@ -697,15 +697,15 @@ TEST_F(TrajectoryPlannerTest, velocity_trajectory_curved_path_low_acceleration)
...
     {
-        const float element_speed = sqrt(pow(velocity_trajectory.x_velocity[i], 2) +
-                                         pow(velocity_trajectory.y_velocity[i], 2));
+        const float element_speed = sqrtf(powf(velocity_trajectory.x_velocity[i], 2) +
+                                          powf(velocity_trajectory.y_velocity[i], 2));
         EXPECT_FLOAT_EQ(position_trajectory.linear_speed[i], element_speed);
 
         EXPECT_FLOAT_EQ(
-            direction_unit_vectors[i].x() * position_trajectory.linear_speed[i],
+            (float)direction_unit_vectors[i].x() * position_trajectory.linear_speed[i],
```
Comment (MathewMacDougall) [inline src/firmware/app/control/trajectory_planner_test.cpp]: nit: we prefer `static_cast` in all `.cpp` files

File: src/firmware/app/world/dribbler_test.cpp [hunk 3]
```diff
@@ -24,7 +24,7 @@ class DribblerTest : public testing::Test
 
     static void setRequestedRpm(uint32_t rpm)
     {
-        requested_rpm = rpm;
+        requested_rpm = (float)rpm;
```
Comment (MathewMacDougall) [inline src/firmware/app/world/dribbler_test.cpp]: nit: we prefer `static_cast` in all `.cpp` files

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]
God's work :heart_eyes: just a few nits about casting

File: src/firmware/app/control/trajectory_planner_impl_test.cpp (hunk 1 above)
Comment (jonathanlew) [reply src/firmware/app/control/trajectory_planner_impl_test.cpp]: fixed

**Review by MathewMacDougall — APPROVED** [review]
:fire: :100: :pray:

**Review by kelvinkoon — APPROVED** [review]
"this was not fun"
Thank you for doing this.

**Review by akhilveeraghanta — APPROVED** [review]
⚠️

**Pull request merged**